blockchain_storage/src/store.rs

```diff
@@ -31,6 +36,7 @@ mod well_known {
 
     declare_entry!(StoreVersion: u32);
     declare_entry!(BestBlockId: Id<Block>);
+    declare_entry!(UtxosBestBlockId: Id<Block>);
```

Comment:
Because the utxo view may or may not update its internal state to reflect the true best block. Ideally, they would work in tandem. But in practice, they're not the same thing. In other words, the utxo storage should be able to check whether it's in sync with the best chain.